#       modified:   core/jni/android_view_HardwareRenderer.cpp
diff --git a/frameworks/base/core/jni/android_view_HardwareRenderer.cpp b/frameworks/base/core/jni/android_view_HardwareRenderer.cpp
index 479fbe2..93468cc 100644
--- a/frameworks/base/core/jni/android_view_HardwareRenderer.cpp
+++ b/frameworks/base/core/jni/android_view_HardwareRenderer.cpp
@@ -124,12 +124,14 @@ static void android_view_HardwareRenderer_beginFrame(JNIEnv* env, jobject clazz,
     eglBeginFrame(display, surface);
 }
 
+#ifndef STE_HARDWARE
 static jlong android_view_HardwareRenderer_getSystemTime(JNIEnv* env, jobject clazz) {
     if (uirenderer::Extensions::getInstance().hasNvSystemTime()) {
         return eglGetSystemTimeNV();
     }
     return systemTime(SYSTEM_TIME_MONOTONIC);
 }
+#endif
 
 #endif // USE_OPENGL_RENDERER
 
@@ -158,9 +160,10 @@ static JNINativeMethod gMethods[] = {
     { "nLoadProperties",        "()Z",   (void*) android_view_HardwareRenderer_loadProperties },
 
     { "nBeginFrame",            "([I)V", (void*) android_view_HardwareRenderer_beginFrame },
-
+#ifndef STE_HARDWARE
     { "nGetSystemTime",         "()J",   (void*) android_view_HardwareRenderer_getSystemTime },
 #endif
+#endif
 
     { "nSetupShadersDiskCache", "(Ljava/lang/String;)V",
             (void*) android_view_HardwareRenderer_setupShadersDiskCache },
